docs/src/documentation/build_pipeline.md

```diff
@@ -1,88 +1,158 @@
+Similar to `required_libs.json` this file contains information which tools are required to develop rex on your platform.
 
-Additionally, Rex Engine has a dependency on 2 smaller Rex modules, namely Rex Stl and Rex Py. like the other dependencies, these are downloaded as well, but if you want to specifically update these as well you must pass either `-update_rsl` or `-update_rexpy` to update rex stl or rex py respectively.
+Some examples for these tools are:
+- compiler
+- linker
+- ninja
+- sharpmake
+- ...
 
-The reason these are not marked submodules at the moment is because they regularly get updates which is too much of a hassle to get properly working with git over multiple branches.
+More on these tools later.
+
+### **required_externals.json**
+For those familiar with Git, this file acts as a description file for the submodules of Rex. 
+
+The reason why they're not stored in a submodule file using Git functionality to clone these dependencies is because it's much faster to store these dependencies in a repository somewhere and manually track which version is needed by your branch.
 
 Some Examples:
 ```sh
 # Will do an initial setup.
 # Updating non-cached paths and cached paths that no longer exist
 py _setup.py
 
-# Same as initial setup but will explicitly update rsl as well.
-py _setup.py -update_rsl
-
-# Same as initial setup but will explicitly update rexpy as well.
-py _setup.py -update_rexpy
-
-# Same as initial setup but will explicitly update rsl and rexpy as well.
-py _setup.py -update_rsl -update_rexpy
+# Same as initial setup but will clean intermediates first
+py _setup.py -clean
 ```
 
 ## Generation
-Rex Engine uses a custom [Sharpmake](https://github.com/RisingLiberty/Sharpmake) implementation to generate ninja files. All you have to do is call `_generate.py`. A visual studio solution generation is also supported for debugging purposes. Such a solution can be generated by passing in `-vs` argument to `_generate.py`
+Rex Engine uses a custom [Sharpmake](https://github.com/RisingLiberty/Sharpmake) for its generation. 
+
+### Sharpmake
+Sharpmake is a build tool, written in C# and who's build scripts are also written in C#. We chose sharpmake as the build tool for Rex due to its extended debug possibilities. Because the build scripts themselves are written in C# this allows you to debug them in an IDE with breakpoints, printing, callstack tracing, ...
+
+Another benefit of using C# is it allows you to run actual C# code while you're generating, giving almost limitless possibilities of what you do during generation, this is much harder to do in other build tools (eg. CMAKE, Sharpmake, NMake, ..)
+
+#### **Rex Additions**
+The vanilla version of Sharpmake is very powerful, it is however mainly used for `Visual Studio` which is very focussed on `Windows`. It does support `XCode` projects among other things, but we didn't want to have different project files on different platforms. This would also make CI much harder. 
```

Comment:
Is this a problem that we don't support XCode?
We have never tested anything on XCode IIRC

Reply:
it's just an example, other project types are also supported, but we want 1 generic one

Reply:
Can you make it clear that it's an example, eg: for example even though we do not support XCode ...

Reply:
Sure